semgrep/semgrep/core_runner.py

```diff
@@ -426,9 +426,14 @@ def handle_regex_patterns(
             ]
         except re.error as err:
             raise SemgrepError(f"invalid regular expression specified: {err}")
-        re_fn = functools.partial(get_re_matches, patterns_re)
-        with multiprocessing.Pool(self._jobs) as pool:
-            matches = pool.map(re_fn, targets)
+
+        if len(targets) > 1:
```

Comment:
It looks like you're assuming that we never test with more than one file? IMO that's a bit fragile. Would it make more sense to set a TESTING global and check against that?

Reply:
Good catch! Although I did slightly better than a global.